src/routes/shopping_cart.py

```
@@ -0,0 +1,126 @@
+
+    :raises HTTPException:
+        - 404 if movie not found
+        - 400 if movie is already in cart or already purchased
+    """
+    service = ShoppingCartService(db)
+    try:
+        await service.add_movie_to_cart(current_user, movie_id)
+        cart = await service.get_cart(current_user)
+        return CartResponse.from_orm(cart)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)